### PR TITLE
Fix `get_json_object` producing wrong output with wildcard paths

### PR DESCRIPTION
**Pull request by ttnghia**
The output of `get_json_object` is wrong when the json path has a wildcard following by a name instruction. In such situations, the early return statement in the code path processing name instruction may cause the entire output row to be a null, instead of ignoring the child objects that do not contain the desired field.

This fixes the issue by removing the early return statement mentioned above.

Closes https://github.com/NVIDIA/spark-rapids/issues/11172.

### CODE REVIEW

**Comment by ttnghia** [comment]
build

File: src/main/cpp/src/get_json_object.cu [hunk 1]
```diff
@@ -488,10 +488,10 @@ __device__ bool evaluate_path(json_parser& p,
                 return false;
               }
             }
-            ctx.task_is_done = true;
-          } else {
-            return false;
           }
+          // Mark task is done regardless whether the expected child was found.
+          // If it was not found, the root context should eventially still not be dirty.
```
Comment (ttnghia) [inline src/main/cpp/src/get_json_object.cu:493]:
```suggestion
```

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
build

**Review by res-life — APPROVED** [review]
LGTM

**Comment by res-life** [comment]
Good catch!

**Pull request merged**